timestamp-impl/src/main/java/com/palantir/timestamp/PersistentUpperLimit.java

```diff
@@ -53,8 +58,10 @@ public long get() {
     }
 
     public synchronized void increaseToAtLeast(long minimum) {
-        if(cachedValue < minimum) {
+        if (cachedValue < minimum) {
             store(minimum);
+        } else {
+            log.trace("Not storing upper limit of {}, as the cached value {} was higher.", minimum, cachedValue);
```

Comment:
might want to do a single volatile read of `cachedValue` to a local as this is a second volatile read though it should be safe since we're syncing on `this`.